app/models/manageiq/providers/hawkular/datawarehouse_manager.rb

```diff
@@ -36,20 +41,32 @@ def self.verify_ssl_mode
     end
 
     # Hawkular Client
-    def self.raw_connect(hostname, port, token, alerts = false)
-      client = alerts ? ::Hawkular::Alerts::AlertsClient : ::Hawkular::Metrics::Client
-      client.new(
+    def self.raw_connect(hostname, port, token, type = :alerts)
+      raise ArgumentError, "Client not found for #{type}" unless CLIENT_KLASSES.keys.include?(type)
+      CLIENT_KLASSES[type].new(
         URI::HTTPS.build(:host => hostname, :port => port.to_i).to_s,
         { :token => token },
         { :tenant => '_system', :verify_ssl => verify_ssl_mode }
       )
     end
 
     def connect(options = {})
-      @client ||= self.class.raw_connect(hostname,
-                                         port,
-                                         authentication_token('default'),
-                                         options[:alerts])
+      raise ArgumentError, "Client not found for #{type}" unless CLIENT_KLASSES.keys.include?(options[:type])
+      @clients ||= {}
+      @clients[options[:type]] ||= self.class.raw_connect(
+        hostname,
+        port,
+        authentication_token('default'),
+        options[:type]
+      )
+    end
+
+    def alerts_client
```

Comment:
Since you are already doing that, why not do it with metaprogramming? That way you don't need to add another method in case another client is added.

```ruby
CLIENT_CLASSES.keys.each do |type|
  define_method("#{type}_client") do
    connect(:type => type)
  end
end
```